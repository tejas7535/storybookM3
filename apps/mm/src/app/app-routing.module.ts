/* eslint-disable @nx/enforce-module-boundaries */
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { LegalRoute } from '@schaeffler/legal-pages';

import { LanguageGuard } from './guards/language.guard';

export enum RoutePath {
  HomePath = '',
  LegalPath = 'legal',
}

export const appRoutePaths: Routes = [
  {
    path: LegalRoute,
    loadChildren: () =>
      import('@schaeffler/legal-pages').then((m) => m.LegalModule),
    canActivate: [LanguageGuard],
  },
  {
    path: RoutePath.HomePath,
    loadComponent: () =>
      import('./home/home.component').then((m) => m.HomeComponent),
    canActivate: [LanguageGuard],
  },

  {
    path: '**',
    redirectTo: RoutePath.HomePath,
  },
];

@NgModule({
  imports: [RouterModule.forRoot(appRoutePaths)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
