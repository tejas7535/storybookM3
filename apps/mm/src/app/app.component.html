<ng-container *transloco="let t">
  <schaeffler-header
    *ngIf="!embedded"
    [platformTitle]="title"
    [link]="'/'"
    class="h-full"
    [toggleEnabled]="true"
    (toggle)="sidebar.toggle()"
  >
    <mm-sidebar #sidebar>
      <ng-container *ngTemplateOutlet="sidebar_content"></ng-container>
    </mm-sidebar>
  </schaeffler-header>

  <div *ngIf="embedded" class="h-full w-full fixed">
    <div class="fixed z-10 bg-white">
      <button
        (click)="sidebar.toggle()"
        mat-icon-button
        [attr.aria-label]="t('settings')"
      >
        <mat-icon class="text-icon">menu</mat-icon>
      </button>
    </div>

    <mm-sidebar #sidebar [embedded]="embedded">
      <ng-container *ngTemplateOutlet="sidebar_content"></ng-container>
    </mm-sidebar>
  </div>

  <ng-template #sidebar_content>
    <div
      [ngClass]="embedded ? 'bg-white' : 'bg-background'"
      class="min-h-full flex flex-col"
    >
      <div
        [ngClass]="embedded ? 'mt-5 pt-5 m-0 p-0' : 'md:m-4 px-4 pb-4'"
        class="flex-auto bg-white"
      >
        <router-outlet></router-outlet>
        <div
          class="mx-auto max-w-screen-md w-full"
          [ngClass]="{ hidden: !isCookiePage }"
        >
          <div class="py-3">
            <button id="ot-sdk-btn" class="ot-sdk-show-settings">
              {{ cookieSettings }}
            </button>
          </div>
          <div class="py-3">
            <div id="ot-sdk-cookie-policy"></div>
          </div>
        </div>
      </div>
      <schaeffler-footer
        *ngIf="!embedded"
        class="flex-shrink-0 bg-white md:bg-background"
        [footerLinks]="footerLinks$ | ngrxPush"
      ></schaeffler-footer>
    </div>
  </ng-template>
</ng-container>
