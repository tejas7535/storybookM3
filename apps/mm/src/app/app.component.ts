/* eslint-disable @nx/enforce-module-boundaries */
import {
  ChangeDetectionStrategy,
  Component,
  inject,
  input,
  OnDestroy,
  OnInit,
} from '@angular/core';
import { toSignal } from '@angular/core/rxjs-interop';
import { Meta, MetaDefinition } from '@angular/platform-browser';
import { NavigationEnd, NavigationSkipped, Router } from '@angular/router';

import {
  combineLatest,
  filter,
  map,
  startWith,
  Subject,
  take,
  takeUntil,
} from 'rxjs';

import { AppShellFooterLink } from '@schaeffler/app-shell';
import {
  EaDeliveryService,
  EaEmbeddedRootComponent,
} from '@schaeffler/engineering-apps-behaviors/utils';
import { LegalPath, LegalRoute } from '@schaeffler/legal-pages';

import packageJson from '../../package.json';
import { MMSeparator } from './core/services';
import { OneTrustMobileService } from './core/services/tracking/one-trust-mobile.service';
import { CalculationSelectionFacade } from './core/store/facades/calculation-selection/calculation-selection.facade';
import { GlobalFacade } from './core/store/facades/global/global.facade';

@Component({
  // eslint-disable-next-line @angular-eslint/component-selector
  selector: 'mounting-manager',
  templateUrl: './app.component.html',
  standalone: false,
  changeDetection: ChangeDetectionStrategy.Default,
})
export class AppComponent
  extends EaEmbeddedRootComponent
  implements OnInit, OnDestroy
{
  private readonly router = inject(Router);
  private readonly meta = inject(Meta);
  private readonly oneTrustMobileService = inject(OneTrustMobileService);
  private readonly globalFacade = inject(GlobalFacade);
  private readonly calculationSelectionFacade = inject(
    CalculationSelectionFacade
  );
  private readonly deliveryService = inject(EaDeliveryService);

  public title = 'Mounting Manager';
  public appVersion = packageJson.version;

  public separator = input<MMSeparator>();
  public outsidemedias = input<boolean>();

  public isCookiePage = false;
  public cookieSettings = this.translocoService.translate(
    'legal.cookieSettings'
  );

  public isBannerOpened = toSignal(this.globalFacade.isBannerOpened$);
  public isStandalone = this.embeddedService.isStandalone;
  public isMobile = this.deliveryService.isMobile;

  public isInitialized$ = this.globalFacade.isInitialized$;
  public destroy$ = new Subject<void>();

  public footerLinks$ = combineLatest([
    this.translocoService.selectTranslate('legal.imprint').pipe(
      map((title) => ({
        link: `${LegalRoute}/${LegalPath.ImprintPath}`,
        title,
        external: false,
      }))
    ),
    this.translocoService.selectTranslate('legal.dataPrivacy').pipe(
      map((title) => ({
        link: `${LegalRoute}/${LegalPath.DataprivacyPath}`,
        title,
        external: false,
      }))
    ),
    this.translocoService.selectTranslate('legal.termsOfUse').pipe(
      map((title) => ({
        link: `${LegalRoute}/${LegalPath.TermsPath}`,
        title,
        external: false,
      }))
    ),
    this.translocoService.selectTranslate('legal.cookiePolicy').pipe(
      map((title) => ({
        link: `${LegalRoute}/${LegalPath.CookiePath}`,
        title,
        external: false,
      })),
      filter(() => !this.isMobile()),
      // eslint-disable-next-line unicorn/no-useless-undefined
      startWith(undefined)
    ),
    this.translocoService.selectTranslate('legal.cookiePolicy').pipe(
      map(
        (title) =>
          ({
            link: 'wrong link',
            title,
            external: false,
            onClick: ($event: MouseEvent) => {
              $event.preventDefault();
              this.oneTrustMobileService.showPreferenceCenterUI();
            },
          }) as AppShellFooterLink
      ),
      filter(() => this.isMobile()),
      // eslint-disable-next-line unicorn/no-useless-undefined
      startWith(undefined)
    ),
  ]).pipe(
    map((links) => links.filter(Boolean)) // Filter out any null values
  );

  public readonly metaTags$ = combineLatest([
    this.translocoService.selectTranslate('meta.title'),
    this.translocoService.selectTranslate('meta.description'),
  ]).pipe(map(([title, description]) => this.makeMetaTags(title, description)));

  public ngOnInit(): void {
    if (this.isStandalone()) {
      this.router.events
        .pipe(
          filter((event) => event instanceof NavigationEnd),
          take(1)
        )
        .subscribe(() => {
          this.globalFacade.initGlobal(this.bearing(), this.separator());
        });
      this.metaTags$.subscribe((tags) => this.meta.addTags(tags));
    } else {
      this.globalFacade.initGlobal(this.bearing(), this.separator());
    }

    window.history.replaceState({ step: 0 }, '', window.location.href);
    this.router.events
      .pipe(
        takeUntil(this.destroy$),
        filter((event) => event instanceof NavigationSkipped)
      )
      .subscribe(() => {
        const { step } = history.state;
        if (step !== undefined) {
          this.calculationSelectionFacade.setCurrentStep(step, true);
        }
      });

    if (
      !this.router.getCurrentNavigation() &&
      !this.router.lastSuccessfulNavigation &&
      this.isStandalone()
    ) {
      this.router.initialNavigation();
    }
  }

  public ngOnDestroy() {
    this.destroy$.next();
    this.destroy$.complete();
  }

  private makeMetaTags(title: string, description: string): MetaDefinition[] {
    const tagFactory = (
      fprefix: string,
      ftitle: string,
      fdescription: string
    ) => [
      { name: `${fprefix + (fprefix === '' ? '' : ':')}title`, value: ftitle },
      {
        name: `${fprefix + (fprefix === '' ? '' : ':')}description`,
        value: fdescription,
      },
    ];
    const prefixes = ['og', 'twitter', ''];

    return prefixes.flatMap((prefix) => tagFactory(prefix, title, description));
  }
}
