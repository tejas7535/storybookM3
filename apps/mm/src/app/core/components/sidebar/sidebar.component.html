<mat-sidenav-container class="h-full" *transloco="let t; read: 'sidebar'">
  <mat-sidenav
    #sidenav
    class="w-full md:w-80 px-4"
    [ngClass]="embedded ? 'pt-12' : 'pt-4'"
  >
    <mat-form-field appearance="outline" class="w-full mt-1">
      <mat-label>{{ t('languageSelectLabel') }}</mat-label>
      <mat-select
        [formControl]="languageSelectControl"
        (selectionChange)="setLanguage($event.value)"
      >
        <mat-option
          *ngFor="let lang of availableLangs; trackBy: trackByFn"
          [value]="lang.id"
          >{{ lang.label }}</mat-option
        >
      </mat-select>
    </mat-form-field>
    <mat-form-field appearance="outline" class="w-full">
      <mat-label>{{ t('decimalSeparatorSelectLabel') }}</mat-label>
      <mat-select
        class="hover:outline-green focus:outline-green"
        [formControl]="separatorSelectControl"
        (selectionChange)="setSeparator($event.value)"
      >
        <mat-option
          *ngFor="let seperator of availableSeparators; trackBy: trackByFn"
          [value]="seperator.id"
          >{{ t(seperator.label) }}</mat-option
        >
      </mat-select>
    </mat-form-field>
  </mat-sidenav>
  <mat-sidenav-content>
    <ng-content></ng-content>
  </mat-sidenav-content>
</mat-sidenav-container>
