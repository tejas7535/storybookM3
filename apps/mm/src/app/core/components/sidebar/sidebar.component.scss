::ng-deep .mat-form-field-appearance-outline .mat-form-field-outline-thick {
  @apply text-primary;
}
