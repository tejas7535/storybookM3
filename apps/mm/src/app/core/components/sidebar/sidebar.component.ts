import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
import { FormControl } from '@angular/forms';
import { MatSidenav } from '@angular/material/sidenav';

import { Subscription } from 'rxjs';

import { AvailableLangs } from '@ngneat/transloco';

import { locales, MMLocales } from '../../services/locale/locale.enum';
import { LocaleService } from '../../services/locale/locale.service';
import { MMSeparator } from '../../services/locale/separator.enum';

interface AvailableSeparators {
  id: string;
  label: string;
}

@Component({
  selector: 'mm-sidebar',
  templateUrl: './sidebar.component.html',
  styleUrls: ['./sidebar.component.scss'],
})
export class SidebarComponent implements OnInit, OnDestroy {
  @ViewChild('sidenav') private readonly sidenav: MatSidenav;

  private readonly subscription = new Subscription();

  private readonly languageSelectControl: FormControl = new FormControl('');
  private readonly separatorSelectControl: FormControl = new FormControl('');

  public availableSeparators: AvailableSeparators[] = [
    { id: ',', label: 'decimalSeparatorComma' },
    { id: '.', label: 'decimalSeparatorPoint' },
  ];

  public availableLangs: AvailableLangs;

  constructor(private readonly localeService: LocaleService) {}

  ngOnInit(): void {
    this.availableLangs = this.localeService.getAvailableLangs();
    this.subscription.add(
      this.localeService.language$.subscribe((language: MMLocales) => {
        this.languageSelectControl.setValue(language);
        this.separatorSelectControl.setValue(
          locales[language].defaultSeparator
        );
      })
    );
    this.subscription.add(
      this.localeService.separator$.subscribe((separator: MMSeparator) => {
        this.separatorSelectControl.setValue(separator);
      })
    );
  }

  ngOnDestroy(): void {
    this.subscription.unsubscribe();
  }

  public setSeparator(separator: MMSeparator): void {
    this.localeService.setSeparator(separator);
  }

  public setLanguage(lang: string): void {
    this.localeService.setLocale(lang as MMLocales);
  }

  public trackByFn(index: number): number {
    return index;
  }

  public toggle(): void {
    this.sidenav.toggle();
  }
}
