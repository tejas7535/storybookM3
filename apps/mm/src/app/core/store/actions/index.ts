export * as CalculationOptionsActions from './calculation-options/calculation-options.actions';
export * as StorageMessagesActions from './storage-messages.action';
