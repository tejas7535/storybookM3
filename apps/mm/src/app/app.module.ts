import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { CoreModule } from './core/core.module';

@NgModule({
  imports: [AppRoutingModule, HttpClientModule, CoreModule],
  bootstrap: [AppComponent],
})
export class AppModule {}
