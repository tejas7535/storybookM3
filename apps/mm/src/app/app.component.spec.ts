import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';

import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { TranslocoTestingModule } from '@ngneat/transloco';
import { provideMockStore } from '@ngrx/store/testing';

import { FooterTailwindModule } from '@schaeffler/footer-tailwind';
import { HeaderModule } from '@schaeffler/header';

import { AppComponent } from './app.component';
import { SidebarComponent } from './core/components/sidebar/sidebar.component';
import { MaterialModule } from './shared/material.module';

describe('AppComponent', () => {
  let component: AppComponent;
  let spectator: Spectator<AppComponent>;

  const createComponent = createComponentFactory({
    component: AppComponent,
    imports: [
      NoopAnimationsModule,
      HeaderModule,
      FooterTailwindModule,
      RouterTestingModule,
      TranslocoTestingModule,

      // TOOD: remove when sidebar component has its module
      ReactiveFormsModule,
      MaterialModule,
    ],
    providers: [provideMockStore()],
    declarations: [AppComponent, SidebarComponent],
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.debugElement.componentInstance;
  });

  test('should create the app', () => {
    expect(component).toBeTruthy();
  });

  it('should set embedded to true if in iframe', () => {
    Object.defineProperty(global, 'window', {
      value: {
        self: 'mockValue',
        top: 'otherMockValue',
      },
    });

    component.checkIframe();
    expect(component.embedded).toBe(true);
  });
});
