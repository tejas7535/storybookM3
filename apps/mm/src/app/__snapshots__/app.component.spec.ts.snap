// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent no inputs getFooterLinks should return the footer links for embedded 1`] = `
Array [
  Object {
    "external": false,
    "link": "legal/imprint",
    "title": "legal.imprint",
  },
  Object {
    "external": false,
    "link": "legal/data-privacy",
    "title": "legal.dataPrivacy",
  },
  Object {
    "external": false,
    "link": "legal/terms-of-use",
    "title": "legal.termsOfUse",
  },
  Object {
    "external": false,
    "link": "legal/cookie-policy",
    "title": "legal.cookiePolicy",
  },
]
`;

exports[`AppComponent no inputs getFooterLinks should return the footer links for native 1`] = `
Array [
  Object {
    "external": false,
    "link": "legal/imprint",
    "title": "legal.imprint",
  },
  Object {
    "external": false,
    "link": "legal/data-privacy",
    "title": "legal.dataPrivacy",
  },
  Object {
    "external": false,
    "link": "legal/terms-of-use",
    "title": "legal.termsOfUse",
  },
  Object {
    "external": false,
    "link": "legal/cookie-policy",
    "title": "legal.cookiePolicy",
  },
]
`;

exports[`AppComponent no inputs makeMetaTags should return the properly formatted tags 1`] = `
Array [
  Object {
    "name": "og:title",
    "value": "title",
  },
  Object {
    "name": "og:description",
    "value": "description",
  },
  Object {
    "name": "twitter:title",
    "value": "title",
  },
  Object {
    "name": "twitter:description",
    "value": "description",
  },
  Object {
    "name": "title",
    "value": "title",
  },
  Object {
    "name": "description",
    "value": "description",
  },
]
`;
